src/plugins/assets.ts

```diff
@@ -0,0 +1,117 @@
+import type { QwikPWAContext } from "../context";
+import type { Plugin } from "vite";
+// import fs from "node:fs/promises";
+// import { join } from "node:path";
+// import { fileURLToPath } from "node:url";
+// import type { DocumentLink } from "@builder.io/qwik-city";
+
+const VIRTUAL = "virtual:qwik-pwa/head";
+const RESOLVED_VIRTUAL = `\0${VIRTUAL}`;
+const DEV_RELOAD_PAGE_NAME = "qwik-pwa:reload-page";
+
+export default function AssetsPlugin(ctx: QwikPWAContext): Plugin {
+  return {
+    name: "qwik-pwa:assets",
+    enforce: "post",
+    resolveId(id) {
+      return id === VIRTUAL ? RESOLVED_VIRTUAL : undefined;
+    },
+    async load(id) {
+      if (id === RESOLVED_VIRTUAL) {
+        const assets = await ctx.assets;
+        return (
+          assets?.resolveHtmlLinks(DEV_RELOAD_PAGE_NAME) ??
+          `export const link = [];
+export const meta = [];
+`
+        );
+      }
+    },
+    async handleHotUpdate({ file, server }) {
+      const assetsGenerator = await ctx.assets;
+      if (await assetsGenerator?.checkHotUpdate(file)) {
```

Comment:
So now we have pwa in dev mode? is that necessary? would love to see your opinion.

Reply:
what if you have to test your custom service worker logic (web push notifications, web shared target api or background sync)? thit hooks is about handling pwa assets on demand and pwa assets changes

Reply:
makes sense.